CMakeLists.txt

```diff
@@ -47,34 +47,115 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 include(CompilerSettings)
 include(FetchInitCPM)
 
-find_package(Eigen3 REQUIRED NO_MODULE)
-
-CPMAddPackage("gh:kavrakilab/nigh#97130999440647c204e0265d05a997dbd8da4e70")
-set(NIGH_INCLUDE_DIRS ${nigh_SOURCE_DIR}/src)
-
-CPMAddPackage("gh:orlp/pdqsort#b1ef26a55cdb60d236a5cb199c4234c704f46726")
-set(PDQSORT_INCLUDE_DIRS ${pdqsort_SOURCE_DIR})
+include(Dependencies)
 
+# Add SIMDxorshift dependency for x86_64 systems
 if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
   CPMAddPackage("gh:lemire/SIMDxorshift#857c1a01df53cf1ee1ae8db3238f0ef42ef8e490")
-  list(APPEND VAMP_SOURCES
-    ${SIMDxorshift_SOURCE_DIR}/src/simdxorshift128plus.c
-    ${SIMDxorshift_SOURCE_DIR}/src/xorshift128plus.c)
-  list(APPEND VAMP_INCLUDES ${SIMDxorshift_SOURCE_DIR}/include)
 endif()
```

Comment:
This should also be handled in `Dependencies.cmake`. You can make `SIMDxorshift_SOURCES` and `SIMDxorshift_INCLUDE` variables that are empty if not on x86, and reference them in the main file.

Reply:
Or better yet - maybe just make a library target for SIMDxorshift (as you do below), but in `Dependencies.cmake`?